tests/test_keys.py

```
@@ -0,0 +1,90 @@
+        tool_fn = _make_tool()
+        with pytest.raises(ToolError, match="key_id"):
+            await tool_fn(action="update")
+
+    async def test_delete_requires_key_id(self, _mock_graphql: AsyncMock) -> None:
+        tool_fn = _make_tool()
+        with pytest.raises(ToolError, match="key_id"):
+            await tool_fn(action="delete", confirm=True)
+
+
+class TestKeysActions:
+    async def test_list(self, _mock_graphql: AsyncMock) -> None:
```

Ruff (0.15.0)

[warning] Fixture `_mock_graphql` without value is injected as parameter, use `@pytest.mark.usefixtures` instead

(PT019)